app/controllers/hyrax/dashboard/collections_controller.rb

```diff
@@ -82,42 +78,30 @@ def edit
 
       def after_create
         form
-        set_default_permissions
         # if we are creating the new collection as a subcollection (via the nested collections controller),
         # we pass the parent_id through a hidden field in the form and link the two after the create.
-        link_parent_collection(params[:parent_id]) unless params[:parent_id].nil?
         respond_to do |format|
-          Hyrax::SolrService.commit
           format.html { redirect_to edit_dashboard_collection_path(@collection), notice: t('hyrax.dashboard.my.action.collection_create_success') }
           format.json { render json: @collection, status: :created, location: dashboard_collection_path(@collection) }
         end
       end
 
-      def after_create_error
-        form
+      def after_create_error(err_msg:)
         respond_to do |format|
           format.html { render action: 'new' }
-          format.json { render json: @collection.errors, status: :unprocessable_entity }
+          format.json { render json: err_msg, status: :unprocessable_entity }
         end
       end
 
       def create
-        # Manual load and authorize necessary because Cancan will pass in all
-        # form attributes. When `permissions_attributes` are present the
-        # collection is saved without a value for `has_model.`
-        @collection = ::Collection.new
-        authorize! :create, @collection
-        # Coming from the UI, a collection type gid should always be present.  Coming from the API, if a collection type gid is not specified,
-        # use the default collection type (provides backward compatibility with versions < Hyrax 2.1.0)
-        @collection.collection_type_gid = params[:collection_type_gid].presence || default_collection_type.to_global_id
-        @collection.attributes = collection_params.except(:members, :parent_id, :collection_type_gid)
-        @collection.apply_depositor_metadata(current_user.user_key)
-        @collection.visibility = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE unless @collection.discoverable?
-        if @collection.save
+        authorize! :create_collection_of_type, collection_type
+        begin
+          create_form_value(user: current_user,
+                            collection_type_gid: params[:collection_type_gid],
+                            parent_id: params[:parent_id])
           after_create
-          add_members_to_collection unless batch.empty?
-        else
-          after_create_error
+        rescue Wings::Valkyrie::Persister::FailedSaveError => ex
```

Comment:
Is there a generic Valkyrie error that all persisters raise when save fails?  Having a Wings error here will cause problems when switching to Postgres or other adatper.

Reply:
what are you trying to rescue from? is there any reason we wouldn't want to run `after_create_error` on all uncaught `StandardError`s raised during execution of the main block?

Reply:
This went away because the exception is rescued in the `save` transaction and it rescues `StandardError`.